src/phases.rs

```diff
@@ -52,6 +54,87 @@ where
     }
 }
 
+impl<T> AddAssign<PhasesOpt<T>> for PhasesOpt<T>
+where
+    T: Add<T, Output = T> + Copy,
+{
+    fn add_assign(&mut self, rhs: PhasesOpt<T>) {
+        *self = *self + rhs;
+    }
+}
+
+impl<T> Sub<PhasesOpt<T>> for PhasesOpt<T>
+where
+    T: Sub<T, Output = T>,
+{
+    type Output = Self;
+
+    fn sub(self, rhs: Self) -> Self {
+        Self(
+            sub_opt(self.0, rhs.0),
+            sub_opt(self.1, rhs.1),
+            sub_opt(self.2, rhs.2),
+        )
+    }
+}
+
+impl<T> SubAssign<PhasesOpt<T>> for PhasesOpt<T>
+where
+    T: Sub<T, Output = T> + Copy,
+{
+    fn sub_assign(&mut self, rhs: PhasesOpt<T>) {
+        *self = *self - rhs;
+    }
+}
+
+impl<T> Mul<Float> for PhasesOpt<T>
+where
+    T: Mul<T, Output = T> + From<Float> + Into<Float>,
+{
+    type Output = Self;
+
+    fn mul(self, rhs: Float) -> Self {
+        Self(
+            mul_opt(self.0, rhs),
+            mul_opt(self.1, rhs),
+            mul_opt(self.2, rhs),
+        )
+    }
+}
+
+impl<T> MulAssign<Float> for PhasesOpt<T>
+where
+    T: Mul<T, Output = T> + From<Float> + Into<Float> + Copy,
+{
+    fn mul_assign(&mut self, rhs: Float) {
+        *self = *self * rhs;
+    }
+}
+
+impl<T> Div<Float> for PhasesOpt<T>
+where
+    T: Div<Float, Output = T> + From<Float> + Into<Float>,
+{
+    type Output = Self;
+
+    fn div(self, rhs: Float) -> Self {
+        Self(
+            div_opt(self.0, rhs),
```

Comment:
muls are faster than divs so it might be more efficient to turn this into self * (1.0/rhs)

Reply:
I think the intent is clearer here without trying to optimise (perhaps the compiler will optimise also?). 

Also, if we're going to optimise then we should optimise the underlying FixedPoint type, no?